File: in_Python/0520 Detect Capital.py
class Solution(object):
    def isC(self,c):
        if c >= 'A' and c <= 'Z':
            return True
        return False
    
    def detectCapitalUse(self, word):
        """
        :type word: str
        :rtype: bool
        """
        first_letter_cap = self.isC(word[0])
        total = 0
        if first_letter_cap:
            total += 1
        for i in range(1,len(word)):
            cur = self.isC(word[i])
            if cur: total += 1
        if total == len(word) or (first_letter_cap and total == 1) or total == 0:
            return True
        else:
            return False
            
        

File: in_Java/0520 Detect Capital.java
class Solution {
    public boolean detectCapitalUse(String word) {
        char[] ch = word.toCharArray();
        boolean f = isC(ch[0]);
        int n = 0;
        for(char c : ch){
            if(isC(c)) n++;
        }
        if(n == 0 || n == word.length()) return true;
        else if(n == 1 && f) return true;
        else return false;
    }
    private boolean isC(char c){
        return c >= 'A' && c <= 'Z';
    }
}